src/github.com/getlantern/balancer/balancer.go

```diff
@@ -63,6 +73,13 @@ func (b *Balancer) OnRequest(req *http.Request) {
 // either manages to connect, or runs out of dialers in which case it returns an
 // error.
 func (b *Balancer) Dial(network, addr string) (net.Conn, error) {
+	lastDialTime := b.lastDialTime.Load().(time.Time)
+	idled := time.Since(lastDialTime)
+	if idled > RecheckAfterIdleFor {
+		log.Debugf("Balancer idled for %s, start checking all dialers", idled)
+		b.checkDialers()
```

Comment:
Ha I'm just too lazy to exclude current dialer from the list. `Dial` may also use multiple dialers if first one fails.